installer/scripts/postinstall.sh

```diff
@@ -80,19 +78,68 @@ check_python_version() {
+ensure_path_other_permissions() {
+  file_path=$1
+  wanted_permissions_other=$2
+
+  # Will output permissions on octal mode - xyz, e.g. 644
+  file_permissions=$(stat -c %a "${file_path}")
+  # Permissions for owner and group - e.g. 644
+  file_permissions_owner_group=$(echo -n "$file_permissions" | head -c 2)
+  # Permissions for other - e.g. 4
+  file_permissions_others=$(echo -n "$file_permissions" | tail -c 1)
+
+  # NOTE: We re-use existing fs permissions for owner and group
+  if [ "${file_permissions_others}" -ne "${wanted_permissions_other}" ]; then
+    new_permissions="${file_permissions_owner_group}${wanted_permissions_other}"
+    echo "Changing permissions for file ${file_path} from \"${file_permissions}\" to \"${new_permissions}\"."
+    chmod "${new_permissions}" "${file_path}" > /dev/null 2>&1;
+  fi
+
+}
+
+# Function which ensures that the provided file path is not readable by "other" users aka has
+# "0" value for permission in the octal notation. If permissions don't match, we update them
+# and ensure value for the user part is "0".
+# This function can operate on a file or on a directory.
+ensure_path_not_readable_by_others() {
+  file_path=$1
+  ensure_path_other_permissions "${file_path}" "0"
+}
+
 check_python_version
 
-config_owner=`stat -c %U /etc/scalyr-agent-2/agent.json`
-script_owner=`stat -c %U /usr/share/scalyr-agent-2/bin/scalyr-agent-2`
+config_owner=$(stat -c %U /etc/scalyr-agent-2/agent.json)
+script_owner=$(stat -c %U /usr/share/scalyr-agent-2/bin/scalyr-agent-2)
 
 # Determine if the agent had been previously configured to run as a
-# different user than root.  We can determine this if agentConfig.json
+# different user than root.  We can determine this if agent.json
 # has a different user.  If so, then make sure the newly installed files
 # (like agent.sh) are changed to the correct owners.
 if [ "$config_owner" != "$script_owner" ]; then
+  echo "Changing owner for /etc/scalyr-agent-2/agent.json file from $script_owner to $config_owner"
   /usr/share/scalyr-agent-2/bin/scalyr-agent-2-config --set_user "$config_owner" > /dev/null 2>&1;
 fi
 
+# Ensure /etc/scalyr-agent-2/agent.json file is not readable by others
+ensure_path_not_readable_by_others "/etc/scalyr-agent-2/agent.json" "0"
+
+# Ensure agent.d/*.json files are note readable by others
+# NOTE: Most software gives +x bit on *.d directories so we do the same
+ensure_path_other_permissions "/etc/scalyr-agent-2/agent.d" "1"
+
+if [ -d "/etc/scalyr-agent-2/agent.d" ]; then
+  # NOTE: find + -print0 correctly handles whitespaces in  filenames and it's more robust than for,
+  # but it may have cross platform issues. In that case we may need to revert to for.
+  #for config_fragment_path in /etc/scalyr-agent-2/agent.d/*.json; do
+  find /etc/scalyr-agent-2/agent.d/ -name "*.json" -print0 | while read -r -d $'\0' config_fragment_path; do
+    ensure_path_not_readable_by_others "${config_fragment_path}" " 0"
```

Comment:
Should there be a space between the quote and the `0`?

Reply:
Yep, this was fixed in a later commit. And I will also add additional guard to the ensure function to validate the argument which is passed in.